app/src/androidTest/java/com/github/sdpteam15/polyevents/view/activity/admin/EventManagementTest.kt

```diff
@@ -427,7 +624,60 @@ class EventManagementTest {
         assertEquals(event!!.getMaxNumberOfSlots(), eventNb.toInt())
         assertEquals(event!!.isLimitedEvent(), true)
         assertEquals(event!!.eventId, eventId)
+    }
+
+
+    @Test
+    fun successfulUpdate() {
+        val startTime = LocalDateTime.now()
+        val endTime = LocalDateTime.now()
+        event = Event(
+            zoneId = listZone[0].zoneId,
+            zoneName = listZone[0].zoneName,
+            eventName = eventName,
+            eventId = eventId,
+            description = eventDesc,
+            limitedEvent = false,
+            startTime = startTime,
+            endTime = endTime
+        )
+        val obs2 = addUpdateListener()
+        val obs = Observable<Boolean>()
+        When(
+            mockedEventDB.getEventFromId(
+                anyOrNull(),
+                anyOrNull(),
+                anyOrNull()
+            )
+        ).thenAnswer {
+            (it.arguments[1] as Observable<Event>).postValue(event!!)
+            obs
```

Comment:
This is not a review, just a question, what's the difference between returning an Observable<Boolean>(true) here and posting true in the observable later as you're doing here?

Reply:
If you return an Observable(true), at the place where this observable is observed, the "observed listener" will be trigger instantaneously. Sometimes, it can be advantageous to trigger the observable at a specific point during the execution to test control flow or something else.

Also, when you test a feature and you want to test both the "true" branch and the "false" branch (in an if-else for instance), you can keep the observable as a variable and post the value for the branch you want to test.

Reply:
Ok very nice! Thanks for the explanation. Never had to use it that way until now, I'll keep that in mind.